IPython/core/oinspect.py

```diff
@@ -821,7 +822,7 @@ def _info(self, obj, oname='', info=None, detail_level=0) -> dict:
                 pass
 
         # Add docstring only if no source is to be shown (avoid repetitions).
-        if ds and out.get('source', None) is None:
+        if ds and not self._source_contains_docstring(out.get('source'), ds):
```

Comment:
This should probably handle `out.get('source')` is none, rather than passing `None` onto the function.

Reply:
We already have to catch all possible parse failures (or parsing into something unexpected) in _source_contains_docstrings (because the source file is re-read when getsource is called, so it could have been modified in between), doesn't seem to be a stretch to also handle the case of None in it.

Reply:
It would be nice to handle that case explicitly, rather than secretly in an `except Exception`

Reply:
This already has to catch at least (SyntaxError, ValueError, TypeError) (syntaxerror because of parse failures, valueerror because `body` may have become more than a single node (and I want to fail in that case too), typeerror because the single node in the body may now be e.g. an Expr node, and in that case ast.get_docstring *will* raise a TypeError.

Distinguishing catching a TypeError that comes from source == None and a TypeError that comes from the source having become a single expression seems like splitting hairs to me.

Reply:
My complaint is that the comment below about the parse failing is technically incorrect - in this case, the parse was never even attempted. Updating the comment is the other way to resolve that

Reply:
done